### PR TITLE
Checking if the parameters are a DictConfig Object

### LINKED ISSUE #2058: Hydra MLFlow Clash
## 🐛 Bug

When using the MLFlow logger with Hydra, because the parameters passed to the LightningModule is a `DictConfig`, the condition in the `logger/base.py` is not met.

https://github.com/PyTorchLightning/pytorch-lightning/blob/8211256c46430e43e0c27e4f078c72085bb4ea34/pytorch_lightning/loggers/base.py#L177

### To Reproduce

Use Hydra and MLFlow together. 

```python
Traceback (most recent call last):
  File "/home/siavash/KroniKare/kwae2/kwae_ma/models/pl_train_segmentation_model.py", line 115, in <module>
    main()
  File "/home/siavash/anaconda3/envs/kwae-ma/lib/python3.7/site-packages/hydra/main.py", line 24, in decorated_main
    strict=strict,
  File "/home/siavash/anaconda3/envs/kwae-ma/lib/python3.7/site-packages/hydra/_internal/utils.py", line 174, in run_hydra
    overrides=args.overrides,
  File "/home/siavash/anaconda3/envs/kwae-ma/lib/python3.7/site-packages/hydra/_internal/hydra.py", line 86, in run
    job_subdir_key=None,
  File "/home/siavash/anaconda3/envs/kwae-ma/lib/python3.7/site-packages/hydra/plugins/common/utils.py", line 109, in run_job
    ret.return_value = task_function(task_cfg)
  File "/home/siavash/KroniKare/kwae2/kwae_ma/models/pl_train_segmentation_model.py", line 111, in main
    trainer.fit(wound_seg_pl)
  File "/home/siavash/anaconda3/envs/kwae-ma/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py", line 765, in fit
    self.single_gpu_train(model)
  File "/home/siavash/anaconda3/envs/kwae-ma/lib/python3.7/site-packages/pytorch_lightning/trainer/distrib_parts.py", line 492, in single_gpu_train
    self.run_pretrain_routine(model)
  File "/home/siavash/anaconda3/envs/kwae-ma/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py", line 843, in run_pretrain_routine
    self.logger.log_hyperparams(ref_model.hparams)
  File "/home/siavash/anaconda3/envs/kwae-ma/lib/python3.7/site-packages/pytorch_lightning/loggers/base.py", line 275, in log_hyperparams
    [logger.log_hyperparams(params) for logger in self._logger_iterable]
  File "/home/siavash/anaconda3/envs/kwae-ma/lib/python3.7/site-packages/pytorch_lightning/loggers/base.py", line 275, in <listcomp>
    [logger.log_hyperparams(params) for logger in self._logger_iterable]
  File "/home/siavash/anaconda3/envs/kwae-ma/lib/python3.7/site-packages/pytorch_lightning/utilities/distributed.py", line 10, in wrapped_fn
    return fn(*args, **kwargs)
  File "/home/siavash/anaconda3/envs/kwae-ma/lib/python3.7/site-packages/pytorch_lightning/loggers/mlflow.py", line 105, in log_hyperparams
    self.experiment.log_param(self.run_id, k, v)
  File "/home/siavash/anaconda3/envs/kwae-ma/lib/python3.7/site-packages/mlflow/tracking/client.py", line 206, in log_param
    self._tracking_client.log_param(run_id, key, value)
  File "/home/siavash/anaconda3/envs/kwae-ma/lib/python3.7/site-packages/mlflow/tracking/_tracking_service/client.py", line 177, in log_param
    _validate_param_name(key)
  File "/home/siavash/anaconda3/envs/kwae-ma/lib/python3.7/site-packages/mlflow/utils/validation.py", line 120, in _validate_param_name
    INVALID_PARAMETER_VALUE)
mlflow.exceptions.MlflowException: Invalid parameter name: ''. Names may be treated as files in certain cases, and must not resolve to other names when treated as such. This name would resolve to '.'
```

### Expected behavior

Check whether the instance if `dict` or `DictConfig` in the given line.

### PR DESCRIPTION
**Pull request by ssakhavi**
#614 
Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## What does this PR do?
Fixes #2058 .

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

_**To be honest, I have no idea how I should go about writing a test for this.**_

### CODE REVIEW

File: pytorch_lightning/loggers/base.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 import torch
 
 from pytorch_lightning.utilities import rank_zero_only
+from omegaconf import DictConfig
```
Comment (Borda) [inline pytorch_lightning/loggers/base.py:12]: `omegaconf` is not mandatory dependency
```suggestion
try:
    from omegaconf import Container
except ImportError:
    Container = None
```

**Comment by ssakhavi** [comment]
@Borda 
The change you proposed caused some of the tests to not pass. Do you know the reason?

**Comment by Borda** [comment]
> The change you proposed caused some of the tests to not pass. Do you know the reason?

you need to propagate the change in your code, like test `Container` instead of `DictConfig`

**Comment by Borda** [comment]
in fact, the solution shall be that `DictConfig` is inherited from `dict`, so no change needed for PL

**Comment by Borda** [comment]
cc: @omry @williamFalcon

**Comment by ssakhavi** [comment]
A question:

If `DictConfig` inherits from `Dict` and the `isinstance` method wasn't working in the crashing version, why should the `MutableMapping` work now? Wouldn't it be the same situation?

**Comment by omry** [comment]
> 
> 
> A question:
> 
> If `DictConfig` inherits from `Dict` and the `isinstance` method wasn't working in the crashing version, why should the `MutableMapping` work now? Wouldn't it be the same situation?

DictConfig inherits from MutableMapping as of OmegaConf 2.0 which comes the Hydra 1.0 release candidate.
If you are using Hydra 0.11.3 you should switch to Hydra 1.0.0rc1 (or newer one they are released) for this to work.

**Pull request merged**